### ISSUE TITLE
Redeploying Subconscious wiped sphere profile

### ISSUE THREAD

**Issue by jsantell** [subconsciousnetwork/subconscious#651]
After a week out of date (and past provision expiry) I built this morning and redeployed; my sphere had nickname `@j2` and a one-line description (95% sure) previously, but since starting with new build, my photo, description are empty and nickname `"unknown"`.

From the logs, this looks suspicious:
```
2023-05-18 14:09:04.918015-0700 Subconscious[55521:2754491] [app] [action] succeedFetchNicknameFromProfile(unknown)
2023-05-18 14:09:04.918552-0700 Subconscious[55521:2754491] [app] Nickname saved: unknown
```

**Comment by bfollington** [comment]
TL;DR: data migration issue, shouldn't affect anyone else going forward.

---

I think I worked out what this is. It's obscure.

Before profiles, we used to store the nickname in AppDefaults (the Apple solution for persistent data like settings). After adding profiles, we had both ways of setting a nickname side by side. My recent PR #639  removed the AppDefaults side of things and relies solely on the profile.

Importantly, that PR ensures the nickname from onboarding is actually persisted in the profile.

For the situation @jsantell encountered above, my hypothesis is that the app had a nickname in AppDefaults but _not_ the user profile and then when deploying the latest version of the app we lost whatever the old value was.

**Comment by bfollington** [comment]
Closing this for now since I believe a one-off, if it re-appear we'll re-open.
